shared/middleware/helpers/file_bucket.rb

```diff
@@ -13,14 +13,37 @@ def allowed_file_type?(extension)
     Sinatra::Base.mime_type(extension)
   end
 
+  def get_manifest(channel_id)
+    manifest_result = get(channel_id, FileBucket::MANIFEST_FILENAME)
+    if manifest_result[:status] == 'NOT_FOUND'
+      return []
+    end
+    JSON.load manifest_result[:body]
+  end
+
   def copy_files(src_channel, dest_channel, options={})
     # copy everything except the manifest
     options[:exclude_filenames] = [MANIFEST_FILENAME]
     result = super src_channel, dest_channel, options
+    # return right away if there are no files in the source project
+    return if result.empty?
+
+    # update dest_manifest with the decorated filenames from the src_manifest
+    # (we need the new version ids from the dest_manifest, otherwise we'd just copy the src_manifest)
+    dest_manifest = JSON.load result.to_json
+    src_manifest = get_manifest(src_channel)
+    src_manifest.each do |src_entry|
+      src_filename = src_entry['filename']
+      dest_manifest.each do |dest_entry|
+        if dest_entry['filename'].downcase == src_filename.downcase
+          dest_entry['filename'] = src_filename
+        end
+      end
+    end
```

Comment:
I _think_ this takes roughly 2n time instead of n^2 time if we do this:
```ruby
case_map = Hash.new.tap do |case_map|
  src_manifest.each do |src_entry|
    case_map[src_entry['filename'].downcase] = src_entry['filename']
  end
end

dest_manifest.each do |dest_entry|
  dest_entry['filename'] = case_map[dest_entry['filename'].downcase]
end
```
It doesn't handle a case yours does though, where the destination manifest contains files that the source manifest doesn't.  Is that something than can happen when using `copy_files`?